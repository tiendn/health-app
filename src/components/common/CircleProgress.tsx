import React from "react";
import styles from "./style.module.scss";
import clsx from "clsx";
import Fonts from "@/theme/fonts";

interface CircleProgressProps {
	date: string;
	radius?: number;
	percentage: number;
}

const CircleProgress: React.FC<CircleProgressProps> = ({ percentage, radius = 90.5, date }) => {
	const size = radius * 2;
	const circumference = 2 * Math.PI * radius;
	const progressOffset = circumference - (percentage / 100) * circumference;

	return (
		<>
			<svg className={styles.circleProgress} width={size} height={size}>
				<circle
					className={styles.progressBar}
					cx={radius}
					cy={radius}
					r={radius}
					strokeWidth="3" // Adjust the stroke width as needed
					strokeDasharray={circumference}
					strokeDashoffset={progressOffset}
				/>
			</svg>
			<div className={clsx(styles.circleProgressContent, Fonts.inter.className)}>
				<span className={styles.circleProgressDate}>{date}</span>
				<span className={styles.circleProgressText}>{percentage}%</span>
			</div>
		</>
	);
};

export default CircleProgress;
