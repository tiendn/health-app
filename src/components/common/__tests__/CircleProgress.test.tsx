import React from "react";
import renderer from "react-test-renderer";
import CircleProgress from "../CircleProgress";

test("Badge renders correctly", () => {
	const tree = renderer.create(<CircleProgress date="2023" percentage={30} />).toJSON();
	expect(tree).toMatchSnapshot();
});
