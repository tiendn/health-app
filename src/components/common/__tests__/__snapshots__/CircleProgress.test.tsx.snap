// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge renders correctly 1`] = `
[
  <svg
    className="circleProgress"
    height={181}
    width={181}
  >
    <circle
      className="progressBar"
      cx={90.5}
      cy={90.5}
      r={90.5}
      strokeDasharray={568.6282702997526}
      strokeDashoffset={398.03978920982684}
      strokeWidth="3"
    />
  </svg>,
  <div
    className="circleProgressContent className"
  >
    <span
      className="circleProgressDate"
    >
      2023
    </span>
    <span
      className="circleProgressText"
    >
      30
      %
    </span>
  </div>,
]
`;
